src/modules/credentials/services/CredentialService.ts

```diff
@@ -543,11 +571,23 @@ export class CredentialService extends EventEmitter {
       )
     }
 
+    // Assert the values in the received credential match the values
+    // that were negotiated in the credential exchange
+    // TODO: Ideally we don't throw here, but instead store that it's not equal.
+    // the credential may still have value, and we could just respond with an ack
+    // status of fail
+    if (credentialRecord.credentialAttributesValues) {
```

Comment:
It seems we could shorten `credentialAttributesValues` just to `credentialAttributes`. Does it contain a list of `CredentialPreviewAttribute` right? I thought that it contains just values from credential preview attributes but the call `convertAttributesToValues(attributesValues)` confused me a little bit. But maybe I'm missing something 🤷‍♂️

Reply:
If we only store the values we lose the mime-types. These are not stored by indy. So you'd need to extract them from one of the exchanged messages.

I think it's a burden the mime-types are not encoded in the values itself like `data:image/png;base64,xxxxxxxx`, but until that's the case (which is the pattern I've seen for W3C credentials) we need to store the preview attributes.

Reply:
But `credentialAttributes` sounds fine. Will update